query/src/api/rpc/exchange/exchange_manager.rs

```diff
@@ -440,7 +457,7 @@ impl QueryCoordinator {
             *shutdown_cause = cause;
         }
 
-        if let Some(executor) = &self.executor {
+        if let Some(executor) = self.executor.take() {
```

Comment:
We should not take the executor. It's using in on_finished.

Reply:
Fixed